### PR TITLE
Subtitle translator

### PR DESCRIPTION
**Pull request by atamariya**
A standalone SRT subtitle translator based on Google translator. Upload the SRT file, choose appropriate language for translation, edit translation, add time offset to SRT and finally save the file.

### CODE REVIEW

**Comment by SubJunk** [comment]
This looks great and it's cool to see AngularJS being used. The version can probably be bumped to 1.7.5 safely, they didn't change much between 1.6 and 1.7.

I'll comment in the code too

File: src/main/external-resources/web/subs.html [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+	<div class="skiptranslate">
+		<h1>Subtitle translator</h1>
+		<div id="google_translate_element"></div>
+	</div>
+	<div class="skiptranslate">
+		<form>
+			<div class="form-group">
+				<input type="file" id="files" name="files[]" multiple
+					onchange="angular.element(this).scope().uploadFile(this.files)" />
```
[the remark this answers is not in the document]
Comment (atamariya) [reply src/main/external-resources/web/subs.html]: This doesn't work. Combining FileSaver.js with Angular seems to be a bit tricky.

File: src/main/external-resources/web/subs.html [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+	
+	<div class="table-responsive">
+		<table class="table table-bordered table-striped">
+			<thead>
+				<tr class="headerRow">
+					<th width="5%" class="skiptranslate">No.</th>
+					<th width="10%" class="skiptranslate">From</th>
+					<th width="10%" class="skiptranslate">To</th>
+					<th width="35%" class="skiptranslate">Orignal Text</th>
+					<th width="45%" class="noBorder skiptranslate">Translated Text</th>
```
[the remark this answers is not in the document]
Comment (atamariya) [reply src/main/external-resources/web/subs.html]: "No." column was a late addition. Fixed it.

File: src/main/external-resources/web/subs.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (SubJunk) [reply src/main/external-resources/web/subs.html]: @atamariya ok sure, no problem. How will users discover this page? Maybe it could be added to a menu?

**Comment by SubJunk** [comment]
Oh cool, I see you've also added some other things since I tested it so I'll re-test. If you could resolve the conflicts too that would be good, hopefully we can merge this soon :)

**Comment by atamariya** [comment]
Resolved the conflicts.

**Comment by valib** [comment]
@SubJunk have you tested the web feeds and streams support? For me it seems that this PR stopped support for them because I have got errors like

`DEBUG 2019-02-26 23:46:12.890 [DLNAResource resolver thread 4-1] Retrieving http://feeds.feedburner.com/tedtalks_video
ERROR 2019-02-26 23:46:12.890 [DLNAResource resolver thread 4-1] Error in parsing stream: http://feeds.feedburner.com/tedtalks_video
java.lang.NullPointerException: null
	at net.pms.network.HTTPResource.downloadAndSendBinary(HTTPResource.java:222)
	at net.pms.network.HTTPResource.downloadAndSendBinary(HTTPResource.java:196)
	at net.pms.dlna.Feed.parse(Feed.java:109)
	at net.pms.dlna.Feed.resolveOnce(Feed.java:94)
	at net.pms.dlna.DLNAResource.resolve(DLNAResource.java:1591)
	at net.pms.dlna.DLNAResource.syncResolve(DLNAResource.java:1571)
	at net.pms.dlna.DLNAResource.run(DLNAResource.java:1368)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
TRACE 2019-02-26 23:46:12.890 [pool-6-thread-10] End of analysis for TED`

for all web.conf items.

**Pull request merged**